src/Core/Domain/Customer/Group/Command/AddCustomerGroupCommand.php

```diff
@@ -75,4 +112,40 @@ public function showPrice(): bool
     {
         return $this->showPrice;
     }
+
+    /**
+     * @return ShopId[]
+     */
+    public function getShopIds(): array
+    {
+        return $this->shopIds;
+    }
+
+    private function assertIsCorrect()
+    {
+        $this->assertReductionIsEqualOrGreaterThanZero();
+    }
+
+    private function assertReductionIsEqualOrGreaterThanZero()
+    {
+        $this->assertIsPrice($this->reduction);
+    }
+
+    private function assertIsPrice(DecimalNumber $price)
+    {
+        $regexCheck = preg_match('/^[0-9]{1,10}(\.[0-9]{1,9})?$/', (string) $price);
+        if (0 === $regexCheck || false === $regexCheck) {
+            throw new InvalidReductionException();
+        }
+
+        if ($price->isNegative()) {
+            throw new InvalidReductionException();
+        }
+
+        $integerPart = (int) $price->getIntegerPart();
+        $fractionalPart = (int) $price->getFractionalPart();
+        if ($integerPart < 0 || $integerPart > 100 || ($integerPart === 100 && $fractionalPart > 0)) {
+            throw new InvalidReductionException();
```

Comment:
quite complex check (idk what it does), but is it really necessary? Maybe it was only needed when DecimalNumber was not in use?
We had some price fields when migrating product, but we never did this kind of checks. 
If its really necessary, then it would make more sense introducing some Price class and reuse, but I think the DecimalNumber should be enough :thinking:

Reply:
No actually the naming is confusing, this is a reduction percent, that's why it must be between 0 and 100

Reply:
Naming was misleading, simplified the assertions